core/primitives/src/shard_layout.rs

```diff
@@ -0,0 +1,143 @@
+use crate::borsh::maybestd::io::Cursor;
+use crate::types::{AccountId, NumShards};
+use borsh::{BorshDeserialize, BorshSerialize};
+use byteorder::{LittleEndian, ReadBytesExt};
+use near_primitives_core::hash::hash;
+use near_primitives_core::types::ShardId;
+use serde::{Deserialize, Serialize};
+use std::cmp::Ordering::Greater;
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum ShardLayout {
+    V0(ShardLayoutV0),
+    V1(ShardLayoutV1),
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
```

Comment:
Do we actually need to derive `BorshSerialize` and `BorshDeserialize`?

Reply:
Yep you are right, it's not needed.